### PR TITLE
Include pulse_list(s) in AQTCompilerOutput

### LINKED ISSUE #45: Include pulse_list(s) in AQTCompilerOutput

### PR DESCRIPTION
**Pull request by singular-value**
Closes #45

### CODE REVIEW

File: cirq_superstaq/aqt.py [hunk 1]
```diff
@@ -18,17 +18,21 @@ def __init__(
         self,
         circuits: Union[cirq.Circuit, List[cirq.Circuit]],
         seq: Optional["qtrl.sequencer.Sequence"] = None,
+        pulse_lists: Optional[Union[List[List], List[List[List]]]] = None,
```
Comment (vtomole) [inline cirq_superstaq/aqt.py:21]: Just wondering: We don't return pulse lists every time?

**Review by vtomole — APPROVED** [review]
LGTM. For another PR: integration test for this.

**Review by richrines1 — APPROVED** [review]
lgtm, but out of curiosity - is there a reason not to define separate `AQTCompilerOutput` and `AQTCompilerOutputs` dataclasses with static attribute names rather than changing attributes based on the input type? it seems like it would clarify some logic and simplify testing a bit, and it's possibly non-ideal for downstream/user-created functions which take in a `AQTCompilerOutput` object to always need to do something like `if hasattr(obj, "circuits"): ...` to avoid throwing an attribute error

**Comment by vtomole** [comment]
I prefer this because we have one `CompilerOutput` type. We can add getters to `AQTCompilerOutput` that return `None` instead of failing on the `hasattr` calls.

**Comment by singular-value** [comment]
@vtomole the pulse list is a list of list of qtrl sequences. Since we don't have qtrl (not open source) on github, we won't be able to do a sensible integration check for it. Ditto for why we don't always return pulse_list(s). If qtrl is not installed in the user's environment, then we set pulse_list(s) to None.

**Comment by singular-value** [comment]
@richrines1 I slightly favor the current approach here but definitely see where you're coming from.
For #agility let's stick with this for now, but we can revisit if/when there is a compelling reason to switch to having two classes/types.

**Comment by vtomole** [comment]
> Since we don't have qtrl (not open source) on github, we won't be able to do a sensible integration check for it
 
Ah, this is an edge case for where we could add an internal daily integration test.

**Comment by richrines1** [comment]
@singular-value @vtomole sounds good, I can definitely see it both ways. In the meantime it might be nice to add a property or method to check instead of having to use `hasattr`

**Comment by singular-value** [comment]
Addressed in #48
